presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -1457,6 +1462,93 @@ protected Node aggregateResult(Node aggregate, Node nextResult)
         throw new UnsupportedOperationException("not yet implemented");
     }
 
+    private static String decodeString(SqlBaseParser.UnicodeStringContext context)
+    {
+        final char delimiter;
+        if (context.UESCAPE() != null) {
+            String delimiterString = context.BASIC_STRING().getText();
+            if (delimiterString.length() != 3) {
```

Comment:
It is a bit cleaner to call `unquote()` first, then check if the length is 1. That keeps the details of unquoted strings hidden inside the `unquote` method.